compare_locales/resource/from_properties.py

```diff
@@ -0,0 +1,68 @@
+from __future__ import annotations
+
+import re
+from typing import Iterable, List, Optional, Union
+
+from ..parsers import Comment as ParserComment
+from ..parsers import Entity as ParserEntity
+from ..parsers import Junk as ParserJunk
+from ..parsers import Whitespace as ParserWhitespace
+from .elements import (
+    Comment,
+    Entry,
+    Junk,
+    Message,
+    Pattern,
+    PatternMessage,
+    Text,
+    VariableRef,
+)
+
+
+def patternFromPropertiesValue(
+    value: str, variables: Optional[re.Pattern[str]] = None
+) -> PatternMessage:
+    """
+    Compile a .properties value into a PatternMessage data class.
+
+    `variables` may be a compiled regexp identifying a variable placeholder from the value.
+    Its captured groups (if any) are ignored.
+    """
+
+    pattern: Pattern = []
+    prevEnd = 0
+    if variables:
+        for match in variables.finditer(value):
+            start = match.start()
+            if start > prevEnd:
+                pattern.append(Text(value[prevEnd:start]))
+            pattern.append(VariableRef(match[0]))
+            prevEnd = match.end()
+    if len(pattern) == 0:
+        pattern.append(Text(value))
```

Comment:
I think this clause is redundant because `prevEnd` will be `0` if `len(pattern) == 0` so the `elif` would match&work for it too.